src/ArrayHelper.php

```diff
@@ -378,6 +378,141 @@ public static function setValue(array &$array, array|float|int|string|null $key,
         $array[self::normalizeArrayKey(array_shift($keys))] = $value;
     }
 
+    /**
+     * Appends a value to an associative array at the key path specified.
+     * If there is no such key path yet, it will be created recursively and an empty array will be initialized.
+     * If the value exists, it will become the first element of the array
+     *
+     *  ```php
+     *   $array = [
+     *     'key' => [],
+     *   ];
+     * ```
+     *
+     *  The result of:
+     * ```php
+     *  ArrayHelper::addValue($array, ['key', 'in'], 'variable1');
+     *  ArrayHelper::addValue($array, ['key', 'in'], 'variable2');
+     * ```
+     *
+     *  will be the following:
+     *
+     *  ```php
+     *   [
+     *       'key' => [
+     *           'in' => [
+     *               'variable1',
+     *               'variable2',
+     *           ]
+     *       ]
+     *   ]
+     *  ```
+     *
+     * @param array $array The array to append the value to.
+     * @param array|float|int|string|null $key The path of where do you want to append a value to `$array`
+     *  the path can be described by an array of keys. If the path is null then `$value` will be appended to the `$array`.
+     *
+     * @psalm-param ArrayKey|null $key
+     *
+     * @param mixed $value The value to be appended.
+     */
+    public static function addValue(array &$array, array|float|int|string|null $key, mixed $value): void
+    {
+        if ($key === null) {
+            /** @var mixed */
+            $array[] = $value;
+            return;
+        }
+
+        $keys = is_array($key) ? $key : [$key];
+
+        while (count($keys) > 0) {
+            $k = self::normalizeArrayKey(array_shift($keys));
+            if (!isset($array[$k])) {
```

Comment:
```suggestion
            if (!isset($array[$k])) {
```

What if `$array[$k]` is null?